packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -66,6 +74,112 @@ class TextMapPropagator {
     })
   }
 
+  _injectB3 (spanContext, carrier) {
+    carrier[b3TraceKey] = spanContext._traceId.toString('hex')
+    carrier[b3SpanKey] = spanContext._spanId.toString('hex')
+    carrier[b3SampledKey] = spanContext._traceFlags.sampled ? '1' : '0'
+
+    if (spanContext._parentId) {
+      carrier[b3ParentKey] = spanContext._parentId.toString('hex')
+    }
+  }
+
+  _extractSpanContext (carrier) {
+    const context = this._extractContext(carrier)
+
+    if (!context) return null
+
+    if (context.traceFlags.sampled !== false) {
+      return new DatadogSpanContext(context)
+    } else {
+      return new NoopSpanContext(context)
+    }
+  }
+
+  _extractContext (carrier) {
+    const b3 = this._extractB3Headers(carrier)
+    const sampled = this._isSampled(b3)
+    const traceFlags = {
+      sampled
+    }
+
+    if (carrier[traceKey] && carrier[spanKey]) {
+      return {
+        traceId: platform.id(carrier[traceKey], 10),
+        spanId: platform.id(carrier[spanKey], 10),
+        traceFlags
```

Comment:
Even though Datadog headers take precedence, we're still sampling the trace based on what the B3 headers say. Is this expected?

Reply:
This was expected but ended up being a bad idea. I have added a Datadog specific header.